src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java

```diff
@@ -129,13 +149,50 @@ private CompletableFuture<Result<Request>> awaitPickup(Request request) {
             populateHoldShelfExpirationDate(
               request.withPickupServicePoint(servicePoint),
               tenantTimeZone
-            ))
+            ).map(calculatedRequest-> modifyHoldShelfExpirationDateBasedOnExpirationDateManagement(tenantTimeZone, calculatedRequest,
+              requestQueue, originalRequest)))
         );
     } else {
-      return completedFuture(succeeded(request));
+      Request updatedRequest = Request.from(request.asJson());
+      requestQueue.update(originalRequest, updatedRequest);
+
+      return requestRepository.update(request)
+        .thenComposeAsync(result -> result.after(v -> requestQueueRepository.updateRequestsWithChangedPositions(requestQueue)));
     }
   }
 
+  private RequestQueue modifyHoldShelfExpirationDateBasedOnExpirationDateManagement(ZoneId tenantTimeZone, Request calculatedRequest,
```

Comment:
Could maybe shorten the method name. How "set" instead of "modify" and "with" instead of "basedOn" so it would be `setHoldShelfExpirationDateWithExpirationDateManagement`.